scene/gui/tree.cpp

```diff
@@ -1923,6 +1923,9 @@ int Tree::propagate_mouse_event(const Point2i &p_pos, int x_ofs, int y_ofs, bool
 				c = c->next;
 				item_h += child_h;
 			}
+
+			if (!c && !p_mod->get_shift() && !p_mod->get_control() && !p_mod->get_command() && !click_handled && p_button != BUTTON_RIGHT)
```

Comment:
Should this work with BUTTON_MIDDLE?

Reply:
Yep. I just ignore right mouse button in this case because it has popup menu

Reply:
But selection does not work with the MMB, so I don't see why it should with deselection

Reply:
Well, then it will not deselect. So nothing broken after all, right? :)